testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/concurrency/ConcurrencyTest.java

```diff
@@ -51,6 +62,38 @@ public void concurrentTest(KeycloakRunnable... tasks) throws Throwable {
         System.out.println("took " + end + " ms");
     }
 
+    // KEYCLOAK-8141 Verify that no attribute values are duplicated, and there are no locking exceptions when adding attributes in parallell
+    @Test
+    @Ignore
+    public void createUserAttributes() throws Throwable {
+        AtomicInteger c = new AtomicInteger();
+
+        UsersResource users = testRealm().users();
+
+        UserRepresentation u = UserBuilder.create().username("attributes").build();
+        Response response = users.create(u);
+        String userId = ApiUtil.getCreatedId(response);
+
+        UserResource user = users.get(userId);
+
+        concurrentTest((threadIndex, keycloak, realm) -> {
+            UserRepresentation rep = user.toRepresentation();
+            rep.singleAttribute("a-" + c.getAndIncrement(), "value");
+            user.update(rep);
+        });
+
+        UserRepresentation rep = user.toRepresentation();
+
+        // Number of attributes should be equal to created attributes, or less (concurrent requests may drop attributes added by other threads)
+        assertTrue(rep.getAttributes().size() <= c.get());
```

Comment:
`assertThat(rep.getAttributes(), Matchers.hasSize(lessThanOrEqualTo(c.get())` has better error reporting.

Reply:
Nitpick - not changing

Reply:
Not a nitpick.

Error reporting with `assertThat` shows the list items which is useful for bug investigation, especially in case of hard-to-reproduce concurrent tests.
Current variant only fails without any further details